Automattic/simplenote-macos
# Toolbar: Search Button

### Details:
As of now, the Search Bar will no longer be visible at all times. Instead, we'll display a Search Action, which (on click) will reveal the Search Bar.

@eshurakov WDYT?
Thanks in advance!!

Ref. #718 

---

### Test: Empty State
1. Launch Simplenote and click over the top right 🔍 icon

- [x] Verify the Search Bar is revealed
- [x] Verify that clicking back into the Editor causes the Search Bar to collapse
- [x] Verify that clicking 🔍 again, and clicking over any other note **causes the bar to collapse** as well
- [x] Verify this **also works super in RTL!**

### Test: Search
1. Launch Simplenote and click over the top right 🔍 icon
2. Enter any keyword

- [x] Verify the Notes List gets filtered
- [x] Verify that clicking over any other Note **does not collapse** the Search Bar
- [x] Verify that clicking over the Editor **does not collapse** the Search Bar.

### Test: Toolbar
1. Launch Simplenote and click over the top right 🔍 icon

- [x] Verify that clicking over the ℹ️  Toolbar Button **does not collapse** the Search Bar
- [x] Verify that clicking over the `(...)` Toolbar Button **does not collapse** the Search Bar

### Test: Tags
1. Launch Simplenote and click over the top right 🔍 icon
2. Enter any keyword

- [x] Verify that clicking over any other tag collapses the search bar
- [x] Verify that the Tags List is not filtered anymore

### Test: Trash
1. Launch Simplenote and click over the top right 🔍 icon
2. Enter any keyword

- [x] Verify that clicking over **Trash** collapses the Search Bar
- [x] Verify that the Search Toolbar Button 🔍 is not even visible anymore

### Test: Shrotcuts
1. Launch Simplenote
2. Press **Option + CMD + F**

- [x] Verify the Search Bar is revealed
- [x] Verify that pressing Option + CMD + F again does nothing

### Release
`RELEASE-NOTES.txt` was updated in 3e47e1b with:

> New Search Toolbar Action #741

### Demo!
![Demo](https://user-images.githubusercontent.com/1195260/101504831-6c2e3080-3952-11eb-9880-6dcbce91dda7.gif)
